docs_src/design/adr/013-Device-Service-Events-Message-Bus.md

```diff
@@ -0,0 +1,215 @@
+    Retained     = "false"
+    AutoReconnect  = "true"
+    ConnectTimeout = "5" # Seconds
+    # TLS configuration - Only used if Cert/Key file or Cert/Key PEMblock are specified
+    SkipCertVerify = "false"
+    Authmode = "none"
+    Secretpath = "mqtt"
+```
+
+The `PublishTopic` has a placeholder for the `Device Name` which gets replaced with the actual `Device Name`. If the place holder doesn't exist in the configured value, the  `PublishTopic` values  is used as is.
+
+#### Application Services
+
+##### [MessageBus]
+
+Similar to above, the Application Services `MessageBus` configuration will change to allow for secure connection to the Message Bus. The `Username` & `Password` have been replaced with `Authmode` & `SecretPath` for secure connections. See **[Secure Connections](#secure-connections)** section below for details.
+
+```toml
+[MessageBus.Optional]
+    # MQTT Specific options
+    # Client Identifiers
+    ClientId ="<app sevice key>"
+    # Connection information
+    Qos          =  "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+    KeepAlive    =  "10" # Seconds (must be 2 or greater)
+    Retained     = "false"
+    AutoReconnect  = "true"
+    ConnectTimeout = "5" # Seconds
+    # TLS configuration - Only used if Cert/Key file or Cert/Key PEMblock are specified
+    SkipCertVerify = "false"
+    Authmode = "none"
+    Secretpath = "mqtt"
+```
+
+##### [Binding]
+
+The `Binding` configuration section will require change for the subscribe topic schema describe in the [Message Bus Topics](#message-bus-topics) section above to filter for Events from specific devices. `SubscribeTopic` will change from a string property containing a single topic to the `SubscribeTopics` string property containing a comma separated list of topics. This allows for the flexibility for the property to be a single topic with the `#` wild card so the Application Service receives all Events as it does today.
+
+```toml
+[Binding]
+Type="messagebus"
+SubscribeTopics="edgex/events/Random-Integer-Device, edgex/events/Random-Boolean-Device"
+```
+
+```toml
+[Binding]
+Type="messagebus"
+SubscribeTopics="edgex/events/#"
+```
+
+### Secure Connections
+
+As stated earlier,  this ADR is dependent on the  [**Secret Provider for All**](TBD) ADR to provide a common Secret Provider for all Edgex Services to access their secrets. Once this is available, the Message Bus connection can be secured via the following configurable authentications modes which follows the implementation for secure MQTT Export and secure MQTT Trigger used in Application Services.
+
+- **none** - No authentication used
+- **usernamepassword** - Username & password authentication. CA Cert used in conjunction if it exists in the Secret Provider. 
+- **clientcert** - Client certificate and key for authentication. CA Cert used in conjunction if it exists in the Secret Provider. 
+- **cacert** - CA Certificate only, i.e. just a TLS enabled connection
```

Comment:
Nice list.  Since clientcert mode requires TLS, does that mean that if a client certificate and key are available, but not a CA certificate, then it uses client TLS but doesn't validate the authenticity of the server?

Reply:
Similar to above, is there a way to make a TLS-enabled connection that doesn't validate the authenticity of the server?

Reply:
Good question. Will verify how we did that in the App SDK.

Reply:
Yes, it uses the uses client for TLS. Validation will depend on if the cert is self-sign or or not. The MQTT options have `SkipVerify` for when using self-signed cert.

Here is the code from SDK to deals with client cert.
```
	case AuthModeCert:
		cert, err = tls.X509KeyPair(secrets.certPemBlock, secrets.keyPemBlock)
		if err != nil {
			return err
		}
		tlsConfig.Certificates = []tls.Certificate{cert}
``

Reply:
@bnevis-i , per your comments in other thread, the Client Cert should be used for Auth as well as TLS? Looks like the App SDK is only using the Client cert for TLS.

Reply:
Note this is all for the client to securing connect...

Reply:
LGTM.